### ISSUE TITLE
Avoid `PossiblyUnusedMethod` on entity's setters when entity is bound to a form

### ISSUE THREAD

**Issue by ThomasLandauer** [psalm/psalm-plugin-symfony#361]
Psalm is reporting:

> ERROR: PossiblyUnusedMethod - src/Entity/Foo.php:759:21 - Cannot find any calls to method App\Entity\Foo::setBar

...even though I have a Symfony form which is bound to `Foo`, and contains a field `bar`.

**Comment by zmitic** [comment]
It is not what you asked for, but I would still recommend to use either the [default callbacks](https://symfony.com/blog/new-in-symfony-5-2-form-mapping-callbacks), or much better [rich-forms-bundle](https://github.com/qossmic/rich-model-forms-bundle/blob/main/docs/mapping.md): 

```php
'write_property_path' => fn (Foo $foo, string $bar) => $foo->setBar($bar),
```

If you use stricter scalars like `non-empty-string`, then you could even do:

```php
'write_property_path' => function (Foo $foo, string $bar) {
    if ('' === $bar) {
        // this exception turns into validation error
        throw new TransformationFailedException(invalidMessage: 'Empty string not allowed');
    }
    // psalm now knows $bar is non-empty-string here
    $foo->setBar($bar); 
}
```

If you install [webmozart/assert](https://github.com/webmozarts/assert), do this first:

```php
class FormAssert extends Assert
{
    /**
     * @psalm-external-mutation-free
     */
    protected static function reportInvalidArgument($message)
    {
        throw new TransformationFailedException(invalidMessage: $message);
    }
}
```

and the above code becomes:

```php
'write_property_path' => function (Foo $foo, string $bar) {
    FormAssert::stringNotEmpty($bar);
    $foo->setBar($bar); 
}
```


It can be PITA to write these, but you get ctrl+click navigation to where getter/setter are used. And if any typehint is wrong, like some data transformer goes unexpected, then you will get validation error instead of 500.
